### DISCUSSION TITLE
Tips for building with OpenColorIO?

### DISCUSSION THREAD

**Discussion by BrianHanke** [OpenImageIO/oiio#4175] [Q&A] [answered]
Howdy all, I'm trying to figure out how to include OpenColorIO when building OIIO on Windows. I'm able to build OIIO without OCIO just fine, and I can also build OCIO on its own.

When I set -DOpenColorIO_ROOT=%OCIO_ROOT% for OIIO, CMake reports that OpenColorIO is found, which sounds good. However, when I go to do the final build in Visual Studio I get a ton of errors, mostly like:

```LNK2038 mismatch detected for '_ITERATOR_DEBUG_LEVEL': value '0' doesn't match value '2' in diff.obj c:\Source\oiio\oiio\build\src\oiiotool\OpenEXR-3_3.lib(ImfBoxAttribute.obj)```

Any tips to point me in the right direction? Thank you!

**Comment by BrianHanke** [comment]
Here are a few more details about my build script. I'm building OpenColorIO third, after boost and zlib, using:

```
git clone https://github.com/AcademySoftwareFoundation/OpenColorIO.git
cd OpenColorIO
cmake -S . -B build -DCMAKE_BUILD_TYPE=Release -DOCIO_BUILD_APPS=OFF -DOCIO_BUILD_PYTHON=OFF -DOCIO_BUILD_TESTS=OFF -DOCIO_BUILD_GPU_TESTS=O -DZLIB_ROOT=%ZLIB_ROOT%
cmake --build build --config Release --target install
```
Then, build the rest of the stuff and end by configuring OIIO with:
```
cmake -DVERBOSE=ON -DCMAKE_BUILD_TYPE=Release -DBoost_USE_STATIC_LIBS=ON -DBoost_NO_WARN_NEW_VERSIONS=ON -DBoost_ROOT=%BOOST_ROOT% -DZLIB_ROOT=%ZLIB_ROOT% -DTIFF_ROOT=%TIFF_ROOT% -DOpenEXR_ROOT=%EXR_ROOT%\dist -DImath_DIR=%EXR_ROOT%\dist\lib\cmake\Imath -DJPEG_ROOT=%JPEG_ROOT% -DPNG_ROOT=%PNG_ROOT% -DOpenColorIO_ROOT=%OCIO_ROOT% -DUSE_PYTHON=0 -DUSE_QT=0 -DBUILD_SHARED_LIBS=0 -DLINKSTATIC=1 -DOIIO_BUILD_TESTS=0 ..
```
That all works, but the final build in Visual Studio fails. Are there any mistakes in my process? Thanks!

**Comment by lgritz** [comment, marked as answer]
That looks basically right. The error you're getting is the one in your first message above?
That's not something OIIO uses directly.

This is sounding suspiciously like you either have multiple versions of OCIO or OpenEXR that are conflicting, or maybe that somehow OCIO was built with a different version of OpenEXR than OIIO used? Something of that flavor?

Hey, what happens if you omit the `-DBUILD_SHARED_LIBS=0 -DLINKSTATIC=1` when you configure OIIO?
Do you have a requirement to have strictly static libraries?

**Reply by BrianHanke** [reply]
It works! 😍 Thanks Larry!

**Comment by lgritz** [comment]
Marking this closed, as it was answered long ago.
